src/main/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactReleaseHistoryScore.java

```diff
@@ -0,0 +1,179 @@
+package com.sap.oss.phosphor.fosstars.model.score.oss;
+
+import static com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures.RELEASED_ARTIFACT_VERSIONS;
+import static java.time.temporal.ChronoUnit.DAYS;
+
+import com.sap.oss.phosphor.fosstars.model.Value;
+import com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures;
+import com.sap.oss.phosphor.fosstars.model.score.FeatureBasedScore;
+import com.sap.oss.phosphor.fosstars.model.value.ArtifactVersion;
+import com.sap.oss.phosphor.fosstars.model.value.ArtifactVersions;
+import com.sap.oss.phosphor.fosstars.model.value.ScoreValue;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.IntSummaryStatistics;
+import java.util.Iterator;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+/**
+ * The scores assesses how well maintained an open source library is (in a specific version).
+ * <ul>
+ *   <li>{@link OssFeatures#RELEASED_ARTIFACT_VERSIONS}</li>
+ * </ul>
+ */
+public class ArtifactReleaseHistoryScore extends FeatureBasedScore {
+
+  /**
+   * Initializes a new score.
+   */
+  public ArtifactReleaseHistoryScore() {
+    super("How the releases history of the artifact is",
+        OssFeatures.RELEASED_ARTIFACT_VERSIONS);
+  }
+
+  @Override
+  public ScoreValue calculate(Value<?>... values) {
+    Value<ArtifactVersions> artifactVersions = find(RELEASED_ARTIFACT_VERSIONS, values);
+
+    if (artifactVersions.isUnknown()) {
+      return scoreValue(0.0, artifactVersions)
+          .makeUnknown()
+          .explain("No versions are found. Hence, no release history score can be calculated")
+          .withMinConfidence();
+    }
+
+    if (artifactVersions.get().size() <= 1) {
+      return scoreValue(0.0, artifactVersions)
+          .explain("Only one version is given. Hence, no release history score can be calculated")
+          .withMinConfidence();
+    }
+
+    ScoreValue scoreValue = scoreValue(7.0, artifactVersions);
+
+    Collection<ArtifactVersion> sortedByReleaseDate = sortByReleaseDate(artifactVersions);
+
+    // check release frequency over time
+    Collection<VersionInfo> versionInfos = createVersionInfos(sortedByReleaseDate);
+    VersionStats stats = createVersionStats(versionInfos);
+
+    if (stats.averageDaysBetweenReleases < 10) {
+      scoreValue.increase(3);
+    } else if (stats.averageDaysBetweenReleases < 30) {
+      scoreValue.increase(2);
+    } else if (stats.averageDaysBetweenReleases < 60) {
+      scoreValue.increase(1);
+    } else if (stats.averageDaysBetweenReleases < 180) {
+      scoreValue.decrease(1);
+    } else if (stats.averageDaysBetweenReleases < 270) {
+      scoreValue.decrease(2);
+    } else if (stats.averageDaysBetweenReleases < 360) {
+      scoreValue.decrease(3);
+    }
+
+    if (stats.releaseCycleTrend < 0) {
+      scoreValue.decrease(-1 * stats.releaseCycleTrend);
+    } else {
+      scoreValue.increase(stats.releaseCycleTrend);
+    }
```

Comment:
Did you mean this?

```suggestion
    scoreValue.increase(stats.releaseCycleTrend);
```

Reply:
If the `releaseCycleTrend` is negative I get a `IllegalArgumentException("Delta can't be negative!")` if I use the `increase` method.
Hence I did it this way.